docs/sections/calculations/errors.md

```diff
@@ -565,346 +222,107 @@ $ verdi calcjob outputcat <pk_number> | less
 
-```{code-block} console
+```{code-block} ipython
 
-$ verdi calcjob outputcat <IDENTIFIER>
+In [9]: parameters_dict = {
+   ...:    "CONTROL": {
+   ...:        "calculation": "scf",
+   ...:    },
+   ...:    "SYSTEM": {
+   ...:        "ecutwfc": 30.,
+   ...:        "ecutrho": 200.,
+   ...:    },
+   ...:    "ELECTRONS": {
+   ...:        "conv_thr": 1.e-14,
+   ...:        'electron_maxstep': 3,
+   ...:    }
+   ...: }
+   ...: builder.parameters = Dict(dict=parameters_dict)
+   ...: calculation = submit(builder)
 
 ```
 
-Use the latter to verify that the energy that you have found in the last step has been extracted correctly from the output file.
-
-:::{tip}
-
-Filter the lines containing the string 'energy', e.g. using `grep`, to isolate the relevant lines.
-
-:::
-
-The results of calculations are stored in two ways: `Dict` objects are stored in the database, which makes querying them very convenient, whereas `ArrayData` objects are stored on the disk.
-The choice of what to store in `Dict` and `ArrayData` nodes is made by the parser of `pw.x` implemented in the [aiida-quantumespresso](<https://github.com/aiidateam/aiida-quantumespresso>) plugin.
-
-The `TrajectoryData` output node is a type of `ArrayData`.
-Once more, use the command `verdi data array show <IDENTIFIER>` to determine the energy obtained from calculation you ran.
-This time you will need to use the identifier of the output `TrajectoryData` of the calculation, with link name `output_trajectory`.
-As you might have realized, the difference now is that the whole series of values of the energy calculated after each ionic step are stored.
-Of course, as we simply ran an `scf` calculation, there is only one value here, but as an exercise, you can restart the calculation with:
+Use `verdi process list -a -p1` to verify that the error code is different now.
+You can check again the outputs and the reports with the tools explained in this section and try to fix it yourself before going on to the next.
 
-```{code-block} python
 
-parameters['CONTROL']['calculation'] = 'vc-relax'
+## Restarting calculations
 
-```
+It turns out that your last calculation did not converge because we stopped the self-consistency iteration cycle before it converged (by only setting a max of 3 cycles).
+In this simple case, you could just re-run the calculation from scratch with a sufficient number of iterations, but for expensive calculations (including a structural relaxation or molecular dynamics), you would like instead to restart your calculation from the previous one to save time.
 
-And once again check the `TrajectoryData` output node with link name `output_trajectory`.
+For this purpose, `CalcJobNode` provides the `get_builder_restart` method.
+Just like the `get_builder` method of the `Code` or of the `Process` class, this creates an instance of the `ProcessBuilder`, but with all inputs already pre-populated from those of the "parent" node.
```

Comment:
I don't think this kind of technical level was mentioned above when first getting a builder? Maybe it could be toned down a bit?
```suggestion
This is similar to the `get_builder` method of the `Code` or of the `Process` class used above, but with all inputs already pre-populated from the latest output of the calculation.
```